Cleanse/ComponentFactory.swift

```diff
@@ -14,17 +14,22 @@ public protocol ComponentFactoryProtocol {
     func build(_ seed: ComponentElement.Seed) -> ComponentElement.Root
 }
 
-public struct ComponentFactory<C: ComponentBase> : ComponentFactoryProtocol {
+public final class ComponentFactory<C: ComponentBase> : ComponentFactoryProtocol {
```

Comment:
Any downsides to changing this to a class from a struct?

Reply:
Reference type now. Meaning it loses the niceties of value types, but this is just a factory responsible for creating objects so it's not like there's going to be a huge difference here.